bin/main.ml

```diff
@@ -901,6 +902,56 @@ let subst =
          )
   , Term.info "subst" ~doc ~man)
 
+let utop =
+  let doc = "Load library in utop" in
+  let man = [ (* TODO *) ] in
+  let go common dir ctx_name args =
+    let utop_target = dir |> Path.of_string |> Utop.utop_exe |> Path.to_string in
+    set_common common ~targets:[utop_target];
+    let log = Log.create () in
+    let (build_system, context, utop_path) =
+      (Main.setup ~log common >>= fun setup ->
+       let context =
+         match ctx_name, setup.contexts with
+         | None, [c] -> c
+         | None, [] -> die "No contexts discovered"
+         | None, ctxs ->
+           die "Specify a --context argument to disambiguate contexts.\
+                @.Available contexts: %s"
+             (ctxs
+             |> List.map ~f:(fun c -> c.Context.name)
+             |> String.concat ~sep:",")
+         | Some name, ctxs ->
+           begin match List.find ctxs ~f:(fun ctx -> ctx.Context.name = name) with
+           | None -> die "Context %s doesn't exist" name
+           | Some c -> c
+           end in
+       let target =
+         match resolve_targets ~log common setup [utop_target] with
+         | [] -> die "no libraries defined in %s" dir
+         | [target] -> target
+         | targets ->
+           match List.find targets ~f:(Path.is_descendant
+                                         ~of_:context.Context.build_dir) with
+           | Some s -> s
+           | None -> die "Couldn't find target for %s in context %s" dir
+                       context.Context.name in
+       do_build setup [target] >>| fun () ->
+       (setup.build_system, context, Path.to_string target)
+      ) |> Future.Scheduler.go ~log in
+    Build_system.dump_trace build_system;
+    Unix.execve utop_path (Array.of_list (utop_path :: args))
```

Comment:
Well, the root issue is that the `execve` system call doesn't exist on Windows, so `Unix.execve` simply fails with `Unix_error (ENOSYS, "execve", "")`